lib/ibmdb.js

```diff
@@ -420,7 +420,9 @@ IBMDB.prototype.updateOrCreate = IBMDB.prototype.save =
           }
 
           if (countData[0]['CNT'] > 0) {
-            stmt = self.buildUpdate(model, data.id, data);
+            var where = {};
+            where[id] = data[id];
```

Comment:
while you are at changing data.id to data[id], there is another occurence in the same function, line 445.